File: hanoiTower.cpp
#include <iostream>

using namespace std;

void hanoiTowerMove(int num, char from, char by, char to);
int main(void)
{
	hanoiTowerMove(3, 'A', 'B', 'C'); //Move from stick A to stick B 
	return 0;
}

void hanoiTowerMove(int num, char from, char by, char to)
{
	if (num == 1)
	{
		cout << "1번째 원판이 " << from << "에서 " << to << "로 이동!!" << endl;
	}
	else
	{
		hanoiTowerMove(num - 1, from, to, by);
		cout << num << "번째 원판이 " << from << "에서 " << to << "로 이동!!" << endl;
		hanoiTowerMove(num - 1, by, from, to);
	}
}

File: YSW_DataStructure/PYTHON_ver/hanoiTower.py
#
#  I try to understand recersion of hanoitower
#
#

def hanoiTowerMove(num: int, froms, by, to):
	if num == 1:
		print("first disk is from ", froms, "to ", to)
	else:
		hanoiTowerMove(num-1, froms, to, by)	
		print(num, "th disk from ", froms, "to ", to)
		hanoiTowerMove(num-1, by, froms, to)

def main():
	hanoiTowerMove(3, 'A', 'B', 'C') # Move from stick A to stick B

if __name__ == "__main__":
	main()